ansible_collections/arista/avd/roles/eos_designs/python_modules/interface_descriptions/__init__.py

```diff
@@ -0,0 +1,165 @@
+            return self.template_var(template_path, template_vars)
+
+        elements = [peer, peer_interface]
+        elements = [str(element) for element in elements if element is not None]
+        return '_'.join(elements)
+
+    def connected_endpoints_port_channel_interfaces(self, peer: str = None, adapter_port_channel_description: str = None) -> str:
+        if template_path := get(self._hostvars, "switch.interface_descriptions.connected_endpoints_port_channel_interfaces"):
+            template_vars = ChainMap({}, self._hostvars)
+            template_vars['peer'] = peer
+            template_vars['adapter_port_channel_description'] = adapter_port_channel_description
+            return self.template_var(template_path, template_vars)
+
+        elements = [peer, adapter_port_channel_description]
+        elements = [str(element) for element in elements if element is not None]
+        return '_'.join(elements)
+
+    def overlay_loopback_interface(self, overlay_loopback_description) -> str:
+        if template_path := get(self._hostvars, "switch.interface_descriptions.overlay_loopback_interface"):
+            template_vars = ChainMap({}, self._hostvars)
+            template_vars['overlay_loopback_description'] = overlay_loopback_description
+            return self.template_var(template_path, template_vars)
+
+        if overlay_loopback_description is not None:
+            return overlay_loopback_description
+
+        if self._default_mpls_overlay_role in ["server", "client"]:
+            return "MPLS_Overlay_peering"
+
+        if self._mpls_lsr is True:
+            return "LSR_Router_ID"
+
+        return "EVPN_Overlay_Peering"
+
+    def vtep_loopback_interface(self) -> str:
+        if template_path := get(self._hostvars, "switch.interface_descriptions.vtep_loopback_interface"):
+            template_vars = ChainMap({}, self._hostvars)
+            return self.template_var(template_path, template_vars)
+
+        return "VTEP_VXLAN_Tunnel_Source"
+
+
+def load_interfacedescriptions(hostvars, templar) -> AvdInterfaceDescriptions:
+    '''
+    Load the python_module defined in `templates.interface_descriptions.python_module`
+    Return the class defined by `templates.interface_descriptions.python_class_name`
+    '''
+    module_path = get(
+        hostvars,
+        'switch.interface_descriptions.python_module',
+        default=DEFAULT_AVD_INTERFACE_DESCRIPTIONS_PYTHON_MODULE
+    )
+    class_name = get(
+        hostvars,
+        'switch.interface_descriptions.python_class_name',
+        default=DEFAULT_AVD_INTERFACE_DESCRIPTIONS_PYTHON_CLASS_NAME
+    )
+    try:
+        cls = getattr(importlib.import_module(module_path), class_name)
+    except ImportError as imp_exc:
+        raise AristaAvdError(imp_exc) from imp_exc
+
+    # if not isinstance(cls, AvdInterfaceDescriptions):
+    if not issubclass(cls, AvdInterfaceDescriptions):
+        raise AristaAvdError(f"{cls} is not an instance of AvdFacts class")
```

Comment:
This error message should probably be more specific - stating clearly that we expect the class to inherit from AvdInterfaceDescriptions not just AvdFacts (otherwise I would be confused if I tried to import a class inheriting from AvdFacts implementing all the different templates).


Also out of curiosity, what happens if a class does not implement properly one of the necessary methods for description that would be needed for a specific AVD repo? Should we add checks to verify some method are present at the minimum?

Reply:
I decided to enforce it by checking if they are a subclass of our default class, so if they only overload one or none of our methods, everything will still work. If they overload our methods with something bad, it is their own fault :)

Reply:
I forgot to update the error message when I moved to this stricter enforcement. Have also updated on ip adddressing